src/annotator/anchoring/text-range.js

```diff
@@ -1,3 +1,21 @@
+/**
+ * Return the combined length of text nodes contained in `node`.
+ *
+ * This is different than `node.textContent` if called on a comment or processing
+ * instruction directly.
+ *
+ * @param {Node} node
+ */
+function nodeTextLength(node) {
+  switch (node.nodeType) {
+    case Node.ELEMENT_NODE:
+    case Node.TEXT_NODE:
+      return /** @type {string} */ (node.textContent).length;
+    default:
```

Comment:
Any other possible quirks with other node types? Are `CDATA section` nodes part of your calculus here, e.g.?

Reply:
Hmm, good question. Handling of CDATA Section nodes is not consistent everywhere and we haven't decided what the behavior _should_ be yet. They _are_ included in an element's `textContent`, but currently `TextPosition.resolve` ignores them and I'm not sure what position anchors do. Fortunately they can only occur in XML documents. I'll file a separate issue for this.

This is a good example of why having one implementation of Range <=> Text position conversion in the application is good!